### ISSUE TITLE
Unknown Python Version Error

### ISSUE THREAD

**Issue by emjun** [pyodide/pyodide#503]
I tried to install the package `requests` via micropip, as described/shown [in the docs.](https://github.com/iodide-project/pyodide/blob/master/docs/pypi.md)
MWE:
My code snippet in Pyodide looks like: 
```
%% py
import micropip
micropip.install('requests')
```

When I run it, I get the following console output and error: 
```
import micropip
micropip.install('requests')

"unknown variable: python_version"
```

[requests](https://2.python-requests.org/en/master/) is a popular package that's entirely written in Python, so i don't think it has to do with trying to load a package that is written in multiple non-Python languages.

Ideas? Suggestions?

**Comment by phorward** [comment]
hi @emjun I think that `requests` in general cannot be used due to missing [socket](https://docs.python.org/3/library/socket.html) support within the browser.

**Comment by emjun** [comment]
Hi @phorward ! 👋  That makes sense. It would be nice to be able to use `requests` for exchanging data between online sources (like if I want to pull data hosted on another website, like data.gov) in Pyodide, a web-first data science platform. So I was hoping maybe there was some workaround I wasn't aware of?

**Comment by phorward** [comment]
> So I was hoping maybe there was some workaround I wasn't aware of?

Sure. You can still use [XHttpRequest](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest) as well as the [fetch-API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) to do so.

I am currently porting a web-app called "ViUR Vi" to Pyodide. It contains also a [network wrapper](https://github.com/viur-framework/vi/blob/pyodide/network.py) that encapsulates XHttpRequests to use with Python, and already runs with Pyodide (the `html5.jseval()`-calls there equal to `js.eval()` calls in the Pyodide API). Maybe you can borrow some things from there.

It would also be an option to implement `requests` (or a subset of it) using the JavaScript fetch-API specific for use with Pyodide. This would be much work, but a great thing. :+1:

Maybe, there are also some other options?

**Comment by lrowe** [comment]
I get this same error when I try to install 'altair'.

**Comment by lrowe** [comment]
Guess this is because micropip does not support the python_version environment marker which altair uses for one of its dependencies: https://github.com/altair-viz/altair/blob/master/requirements.txt#L8

**Comment by hoodmane** [comment]
Fixed by #1483.
